### PR TITLE
fix: enable objectData to be specified as a simple text field

### PR DESCRIPTION
**Pull request by JasonBerry**
https://algolia.atlassian.net/browse/EEX-884

The "simple table" method would be pretty much unusable for customers trying to add objectData. Using a text field will allow customers to set the value based on some variable that they specify.

Also, there were a few too many methods that we were allowing to send objectData, where we only really want it for conversions.

### CODE REVIEW

**Comment by benhinchley** [comment]
@JasonBerry is there a ticket or some extra context you can provide in the PR description here?

File: src/template-parameters.json [hunk 1]
```diff
@@ -479,35 +469,15 @@
...
-          {
-            "paramName": "method",
-            "type": "EQUALS",
-            "paramValue": "viewedObjectIDs"
           }
         ]
       },
       {
-        "type": "SIMPLE_TABLE",
+        "type": "TEXT",
```
Comment (benhinchley) [inline src/template-parameters.json:476]: Forgive my ignorance here, but does a field of type "TEXT" allow the user to provide a variable to set a field within objectData?

For example if I store the `queryID` in a GTM variable, do I have access to that in this field?

**Review by benhinchley — CHANGES_REQUESTED** [review]

File: src/template-parameters.json (hunk 1 above)
Comment (JasonBerry) [reply src/template-parameters.json:476]: You can use literally _any_ variable that you define.

Comment (JasonBerry) [reply src/template-parameters.json:476]: The idea is that the customer would assign this object data info in a JS variable (or something similar) and read it using this field. This seems to be what other GTM tags do in order to read/write more complex data structures.

**Review by benhinchley — APPROVED** [review]

File: src/template-parameters.json [hunk 2]
```diff
@@ -460,16 +460,6 @@
         "simpleValueType": true,
         "type": "TEXT",
         "enablingConditions": [
-          {
```
Comment (jonathaningram) [inline src/template-parameters.json:463]: Was this removal intended? `clickedObjectIDs` for example needs the `objectIDs` field here, see https://www.algolia.com/doc/api-reference/api-methods/clicked-object-ids/?client=javascript

File: src/template-parameters.json [hunk 3]
```diff
@@ -479,35 +469,15 @@
...
       },
       {
-        "type": "SIMPLE_TABLE",
+        "type": "TEXT",
         "name": "objectData",
         "displayName": "Object Data",
-        "help": "A list of data containing additional information about associated objects (maximum of 20).",
+        "help": "A list of <a href=\"https://www.algolia.com/doc/api-reference/api-methods/added-to-cart-object-ids/#method-param-objectdata-2\">data</a> containing additional information about associated objects (maximum of 20).",
         "enablingConditions": [
-          {
```
Comment (jonathaningram) [inline src/template-parameters.json:496]: Similar to the above, if we allow `objectIDs` on a method, I think we should also allow `objectData`. It's just an unfortunate design that these two fields are split apart (instead of something like a single field named `objects`), and so I think we should always pair the two together.

File: src/template-parameters.json (hunk 2 above)
Comment (JasonBerry) [reply src/template-parameters.json:463]: It was. I checked the API and was trying to limit the confusion. We only need revenue info on the conversion events (can you think of why we might need it on the non-conversion events?)  so I thought to remove it.

Comment (jonathaningram) [reply src/template-parameters.json:463]: Don't we pass `objectIDs` to click events? i.e. the object that was clicked? https://www.algolia.com/doc/api-reference/api-methods/clicked-object-ids/?client=javascript#examples

Similarly with view events, the objects that were viewed?

Comment (JasonBerry) [reply src/template-parameters.json:463]: `objectIDs` yes, `objectData` no.

Comment (jonathaningram) [reply src/template-parameters.json:463]: You've removed these from under `objectIDs` though, unless I'm reading it wrong.

File: src/template-parameters.json (hunk 3 above)
Comment (jonathaningram) [reply src/template-parameters.json:496]: Per my comment above, I think you've accidentally removed `objectID` from valid methods above.

My thoughts still stand here (objectID and objectData are the same thing so should always be paired together), but we can probably not make my comment here non-blocking. If users end up asking to send `objectData` on these use cases, we can always add it.

So feel free to resolve this comment.

Comment (JasonBerry) [reply src/template-parameters.json:496]: Ah, good catch. 🤦‍♂️

**Review by jonathaningram — APPROVED** [review]
